meilisearch/tests/index/test_index.py

```diff
@@ -73,60 +93,71 @@ def test_get_or_create_index(self):
         assert len(documents) == 1
         index_2.delete()
         with pytest.raises(Exception):
-            self.client.get_index(index_3).info()
+            self.client.get_index(index_3)
 
     def test_get_or_create_index_with_primary_key(self):
         """Test get_or_create_index method with primary key"""
         index_1 = self.client.get_or_create_index('books', {'primaryKey': self.index_uid4})
         index_2 = self.client.get_or_create_index('books', {'primaryKey': 'some_wrong_key'})
+        assert index_1.primary_key == self.index_uid4
         assert index_1.get_primary_key() == self.index_uid4
+        assert index_2.primary_key is None
         assert index_2.get_primary_key() == self.index_uid4
+        assert index_2.primary_key == self.index_uid4
         index_1.delete()
 
-    def test_index_info(self):
-        """Tests getting an index's info"""
-        index = self.client.get_index(uid=self.index_uid)
-        response = index.info()
+    def test_index_fetch_info(self):
+        """Tests getting the index info"""
+        index = self.client.index(uid=self.index_uid)
+        response = index.fetch_info()
         assert isinstance(response, object)
-        assert response['uid'] == self.index_uid
-        assert response['primaryKey'] is None
-
-    def test_index_info_with_wrong_uid(self):
-        """Tests getting an index's info in MeiliSearch with a wrong UID"""
-        with pytest.raises(Exception):
-            self.client.get_index(uid='wrongUID').info()
+        assert response.uid == self.index_uid
+        assert response.primary_key is None
+        assert response.primary_key == index.primary_key
+        assert response.primary_key == index.get_primary_key()
+
+    def test_index_fetch_info_containing_primary_key(self):
+        """Tests getting the index info"""
+        index = self.client.index(uid=self.index_uid3)
+        response = index.fetch_info()
+        assert isinstance(response, object)
+        assert response.uid == self.index_uid3
+        assert response.primary_key == 'book_id'
+        assert response.primary_key == index.primary_key
+        assert response.primary_key == index.get_primary_key()
 
     def test_get_primary_key(self):
         """Tests getting the primary-key of an index"""
-        index = self.client.get_index(uid=self.index_uid)
+        index = self.client.index(uid=self.index_uid3)
+        assert index.primary_key is None
         response = index.get_primary_key()
-        assert response is None
+        assert response == 'book_id'
```

Comment:
Same as the L125 comment

Reply:
https://github.com/meilisearch/meilisearch-python/pull/175#discussion_r525020400